File: src/main/java/hoard/rxjava2/RxDepositor.java
package hoard.rxjava2;

import hoard.ReactiveStreamDepositor;
import io.reactivex.Completable;
import io.reactivex.Single;

/**
 * A {@link ReactiveStreamDepositor} that converts methods to RxJava 2.x
 * equivalents.
 *
 * @param <T> The type the depositor saves and retrieves.
 * @since 1.0
 */
public interface RxDepositor<T> {
  Completable store(T value);
  Single<T> retrieve();
  Completable delete();
}

File: src/main/kotlin/io/github/roguesdev/hoard/rxjava2/RxDepositor.kt
package io.github.roguesdev.hoard.rxjava2

import io.github.roguesdev.hoard.ReactiveStreamDepositor
import io.reactivex.Completable
import io.reactivex.Single

/**
 * A [ReactiveStreamDepositor] that converts methods to RxJava 2.x
 * equivalents.
 *
 * @param <T> The type the depositor saves and retrieves.
 * @see io.github.roguesdev.hoard.Depositor
 *
 * @since 1.0
</T> */
interface RxDepositor<T> {
    fun store(value: T): Completable
    fun retrieve(): Single<T>
    fun delete(): Completable
    fun exists(): Single<Boolean>
}
